### PR TITLE
fix: Fix cibadge in Readme.

### LINKED ISSUE #3262: Readme CI badge links to badge instead of CI overview
I would expect to see the (currently) failing CI instead of the badge when clicking on it. I guess the link needs to be fixed.

### PR DESCRIPTION
**Pull request by CusiniM**
Trying to address issue  https://github.com/GEOS-DEV/GEOS/issues/3262

I don't know if there is way to link directly to the latest CI job that was run on develop so I have linked it to this https://github.com/GEOS-DEV/GEOS/actions?query=branch%3Adevelop

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [![DOI](https://zenodo.org/badge/131810578.svg)](https://zenodo.org/badge/latestdoi/131810578)
 [![codecov](https://codecov.io/github/GEOS-DEV/GEOS/graph/badge.svg?token=0VTEHPQG58)](https://codecov.io/github/GEOS-DEV/GEOS)
-![CI](https://github.com/GEOS-DEV/GEOS/actions/workflows/ci_tests.yml/badge.svg)
+[![CI](https://github.com/GEOS-DEV/GEOS/actions/workflows/ci_tests.yml/badge.svg)](https://github.com/GEOS-DEV/GEOS/actions?query=branch%3Adevelop)
```
Comment (timokoch) [inline README.md:3]:
```suggestion
[![CI](https://github.com/GEOS-DEV/GEOS/actions/workflows/ci_tests.yml/badge.svg)](https://github.com/GEOS-DEV/GEOS/actions/workflows/ci_tests.yml)
```
I think this shows the workflow corresponding to the badge status

Comment (CusiniM) [reply README.md:3]: that seems to point to the latest CI job that was run which can be on any PR related branch. I think it is more useful to point to the latest job that was run for `develop`

Comment (timokoch) [reply README.md:3]: ok but does the badge know about that?

Comment (timokoch) [reply README.md:3]:
```suggestion
[![CI](https://github.com/GEOS-DEV/GEOS/actions/workflows/ci_tests.yml/badge.svg?branch=develop)](https://github.com/GEOS-DEV/GEOS/actions?query=branch%3Adevelop)
```
I guess like this it does. Ah... if `develop` is the default branch then this is not needed.

**Review by rrsettgast — APPROVED** [review]

**Review by timokoch — APPROVED** [review]

**Pull request merged**